vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -585,12 +586,13 @@ type
                 match hier with
                 | :? IProvideProjectSite as siteProvider when not (IsScript(filename)) ->
                     this.SetupProjectFile(siteProvider, this.Workspace, "SetupNewTextView")
-                | _ when not (IsScript(filename)) ->
+                | h when not (IsScript(filename)) ->
                     let docId = this.Workspace.CurrentSolution.GetDocumentIdsWithFilePath(filename).FirstOrDefault()
                     match docId with
                     | null ->
-                        let fileContents = VsTextLines.GetFileContents(textLines, textViewAdapter)
-                        this.SetupStandAloneFile(filename, fileContents, this.Workspace, hier)
+                        if not (h.IsCapabilityMatch("CPS")) then
```

Comment:
I made this is as narrow as possible.  We could widen this so that we do nothing even if the file is in the project.  However, this will always occur before any events, I believe even before the solution load event, which means we can start intellisense at the earliest possible moment.

I think it's marginal, and so I could be persuaded to widen this.

Reply:
I believe that VS's solution events have fired, but Roslyn's `VisualStudioProjectTracker` hasn't pushed the project to the `Workspace` yet, causing the check above to fail.

With this fix, how/when does the project system get updated for CPS projects?

Can/should we be calling [`AbstractProject.StartPushingToWorkspaceAndNotifyOfOpenDocuments`](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/AbstractProject.cs,830) like Roslyn does when a file is opened?

Reply:
OnDocumentOpened is not invoked for C# when a project is opened with an open file.  It occurs when a File is specifically opened with the open command.

A longer term approach is to move to the MiscellaneousFileWorkspace events for loose files. And remove the need for SetupNewTextView for any of our CPS based files.  But that is a riskier approach, and should probably happen alongside moving to CPS even for our legacy project files, but that has implications for flavoured projects.

Anyway as you can see there is a lot further to go with F# project loading and initialization.  This particular fix is the narrowest fix that addresses the issue and doesn't take back any currently "working scenarios".

Kevin